crates/chain/src/local_chain.rs

```diff
@@ -148,6 +158,39 @@ impl CheckPoint {
     pub fn iter(&self) -> CheckPointIter {
         self.clone().into_iter()
     }
+
+    /// Query for checkpoint at `height`.
+    ///
+    /// Returns `None` if checkpoint at `height` does not exist`.
+    pub fn query(&self, height: u32) -> Option<Self> {
+        self.iter()
+            // optimization to avoid iterating the entire chain if we do not get a direct hit
+            .take_while(|cp| cp.height() >= height)
+            .find(|cp| cp.height() == height)
+    }
+
+    /// Iterate checkpoints over a height range.
+    ///
+    /// Note that we always iterate checkpoints in reverse height order (iteration starts at tip
```

Comment:
:pick: there's no need to start this sentence with `Note that`. It doesn't start at the tip it starts at the highest height within the range.

The implementation looks correct fwiw. I still feel like we are adding methods like this before we actually need them. My bias is to keep it for now since you've done it but we really have to stop adding these kinds of methods.

Reply:
This is needed though, like we're actually using it

Reply:
I see one place it's being used where [as we've discussed](https://github.com/bitcoindevkit/bdk/pull/1380#discussion_r1540554140) it's not necessary.

Reply:
Conversation continues here: https://github.com/bitcoindevkit/bdk/pull/1369#discussion_r1550753230